src/graph_ops.rs

```diff
@@ -24,7 +24,7 @@ pub trait InDegreePartitionOp<Node: Idx> {
 pub trait ForEachNodeOp<Node: Idx> {
     fn for_each_node<T, F>(
```

Comment:
maybe `for_each_node` and `for_each_node_by_partition` or `for_each_node_using_partition` instead of the `Option`?

Reply:
sure :)

Reply:
Do you want it in the same trait or a new trait too?

Reply:
No reason to create a new trait I guess?

Reply:
Ended up making a new trait